### DISCUSSION TITLE
Linear trajectory

### DISCUSSION THREAD

**Discussion by macsierrahe** [petercorke/robotics-toolbox-python#262] [Q&A]
Hello, 
I try to find a function for create linear trajectory with my robot. I have found that we can use some function in Matlab as 'transformtraj', 'trapveltraj' and 'rottraj'. I have found the following example from https://blogs.mathworks.com/student-lounge/2019/11/06/robot-manipulator-trajectory/. My question is : are there the equivalent fonction in python ?  THANK YOU SO MUCH FOR YOUR HELP :3 :*

T0 = trvec2tform([0 0 0]);
Tf = trvec2tform([1 2 3])*eul2tform([pi/2 0 pi/4],'ZYX');
tTimes = linspace(0,1,51);
tInterval = [0 5];

[s,sd,sdd] = trapveltraj([0 1],numel(tTimes));
[T,dT,ddT] = transformtraj(T0,Tf,tInterval,tTimes,'TimeScaling',[s;sd;sdd]);

plotTransforms(tform2trvec(T),tform2quat(T));

**Comment by petercorke** [comment]
sorry to be very slow.

*  `trapveltraj` maps to `trapezoidal`
* `rottraj` can be handled by the `interp` method of `UnitQuaternion` or `SO3`
* `transformtraj` can be handled by the `interp` method of `SE3` or `Twist3`, or the `ctraj` function.
